cpp/test/sparse/preprocess_csr.cu

```diff
@@ -0,0 +1,209 @@
+{
+  cudaStream_t stream = raft::resource::get_cuda_stream(handle);
+  int num_rows        = csr_in.structure_view().get_n_rows();
+  int num_cols        = csr_in.structure_view().get_n_cols();
+  int rows_size       = csr_in.structure_view().get_indptr().size();
+  int cols_size       = csr_in.structure_view().get_indices().size();
+  int elements_size   = csr_in.get_elements().size();
+
+  auto h_rows  = raft::make_host_vector<int, int64_t>(handle, rows_size);
+  auto h_cols  = raft::make_host_vector<int, int64_t>(handle, cols_size);
+  auto h_elems = raft::make_host_vector<float, int64_t>(handle, elements_size);
+
+  auto indptr = raft::make_device_vector_view<T1, int64_t>(
+    csr_in.structure_view().get_indptr().data(), csr_in.structure_view().get_indptr().size());
+  auto indices = raft::make_device_vector_view<T1, int64_t>(
+    csr_in.structure_view().get_indices().data(), csr_in.structure_view().get_indices().size());
+  auto values = raft::make_device_vector_view<T2, int64_t>(csr_in.get_elements().data(),
+                                                           csr_in.get_elements().size());
+  auto rows   = raft::make_device_vector<T1, int64_t>(handle, values.size());
+
+  raft::sparse::convert::csr_to_coo<T1>(
+    indptr.data_handle(), num_rows, rows.data_handle(), rows.size(), stream);
+
+  raft::copy(h_rows.data_handle(), rows.data_handle(), rows.size(), stream);
+  raft::copy(h_cols.data_handle(), indices.data_handle(), cols_size, stream);
+  raft::copy(h_elems.data_handle(), values.data_handle(), values.size(), stream);
+  raft::util::preproc_coo<T1, T2>(
+    handle, h_rows.view(), h_cols.view(), h_elems.view(), results, num_rows, num_cols, tf_idf);
+}
+
+template <typename Type_f, typename Index_>
+struct SparsePreprocessInputs {
+  int n_rows;
+  int n_cols;
+  int nnz_edges;
+};
+
+template <typename Type_f, typename Index_>
+class SparsePreprocessCSR
+  : public ::testing::TestWithParam<SparsePreprocessInputs<Type_f, Index_>> {
+ public:
+  SparsePreprocessCSR()
+    : params(::testing::TestWithParam<SparsePreprocessInputs<Type_f, Index_>>::GetParam()),
+      stream(resource::get_cuda_stream(handle))
+  {
+  }
+
+ protected:
+  void SetUp() override {}
+
+  void Run(bool bm25_on)
+  {
+    cudaStream_t stream = raft::resource::get_cuda_stream(handle);
+
+    int num_rows = pow(2, params.n_rows);
+    int num_cols = pow(2, params.n_cols);
+
+    auto rows    = raft::make_device_vector<Index_, int64_t>(handle, params.nnz_edges);
+    auto columns = raft::make_device_vector<Index_, int64_t>(handle, params.nnz_edges);
+    auto values  = raft::make_device_vector<Type_f, int64_t>(handle, params.nnz_edges);
+    auto mask    = raft::make_device_vector<Index_, int64_t>(handle, params.nnz_edges);
+
+    raft::util::create_dataset<Index_, Type_f>(
+      handle, rows.view(), columns.view(), values.view(), 5, params.n_rows, params.n_cols);
+    int non_dupe_nnz_count = raft::util::get_dupe_mask_count<Index_, Type_f>(
```

Comment:
Declaring the `non_dupe_nnz_count` as `int64_t` might be safer since it is used as `int64_t`  in the following code.